build.rs

```diff
@@ -0,0 +1,16 @@
+fn main() {
```

Comment:
Can we do this without adding a build.rs file just for this purpose?

Reply:
I'm open to suggestions. The reason I added this was because @kubouch gave feedback in [the last PR](https://github.com/nushell/nushell/pull/10013#pullrequestreview-1596828128) that it was unwieldy to repeat

```rust
    #[cfg(all(
        feature = "trash-support",
        not(target_os = "android"),
        not(target_os = "ios")
    ))]
```

in half a dozen places, which I agree with. Another option would be [`cfg_aliases`](https://docs.rs/cfg_aliases/latest/cfg_aliases/), but I opted not to go this route because of other feedback about not adding more external dependencies.

I'm open to anything though, this was really just added in response to other review feedback.